cogs/random_resource.py

```diff
@@ -0,0 +1,47 @@
+import discord
+from discord import app_commands
+from discord.ext import commands
+import random
+
+from api import get_random_resource
+
+
+class RandomResourceCog(commands.Cog):
+    def __init__(self, bot: commands.Bot):
+        self.bot = bot
+
+    @app_commands.command(
+        name="random-resource", description="Send a random WIT resource!"
+    )
+    async def send_random_resource(self, int: discord.Interaction):
+        link_format = "https://unswwit.com/media/"
+        link = ""
+        resources = get_random_resource()
+        resource_type = random.choice(list(resources.keys()))
+        resource = None
+        resource_title = ""
+
+        if resource_type == "publication":
+            resource = random.choice(resources[resource_type]).fields()
```

Comment:
Repeated code can be moved out of if-else statement, you can put this before if-else